### PR TITLE
Warn when setting up a system with local or mem block adapters

### PR DESCRIPTION
**Pull request by arielshaqed**
These are _great_ for quick setup, but should not be used in production.  Warn about it 3 ways:

# UI

On the setup page:
"setup" page says: Block adapter "local" not usable in production

# Terminal

During startup:
```

 ██       ██                           ██                 
░██      ░██                          ░░                  ░██
░██   █  ░██  ██████   ██████ ███████  ██ ███████   █████ ░██
░██  ███ ░██ ░░░░░░██ ░░██░░█░░██░░░██░██░░██░░░██ ██░░░██░██
░██ ██░██░██  ███████  ░██ ░  ░██  ░██░██ ░██  ░██░██  ░██░██
░████ ░░████ ██░░░░██  ░██    ░██  ░██░██ ░██  ░██░░██████░░ 
░██░   ░░░██░░████████░███    ███  ░██░██ ███  ░██ ░░░░░██ ██
░░       ░░  ░░░░░░░░ ░░░    ░░░   ░░ ░░ ░░░   ░░   █████ ░░ 
                                                   ░░░░░  
Using the "local" block adapter.  This is suitable only for testing, but not
for production.
```

# Logs

Also write an error on startup:
```
ERROR  [2021-05-26T09:54:55+03:00]cmd/lakefs/cmd/run.go:112 cmd/lakefs/cmd.glob..func9 Block adapter NOT SUPPORTED for production use  adapter_type=local
```

### CODE REVIEW

File: cmd/lakefs/cmd/run.go [hunk 1]
```diff
@@ -212,6 +219,24 @@ func printWelcome(w io.Writer) {
...
 
+var localWarningBanner = `
+ ██       ██                           ██              	  
+░██      ░██                          ░░                  ░██
+░██   █  ░██  ██████   ██████ ███████  ██ ███████   █████ ░██
+░██  ███ ░██ ░░░░░░██ ░░██░░█░░██░░░██░██░░██░░░██ ██░░░██░██
+░██ ██░██░██  ███████  ░██ ░  ░██  ░██░██ ░██  ░██░██  ░██░██
+░████ ░░████ ██░░░░██  ░██    ░██  ░██░██ ░██  ░██░░██████░░ 
+░██░   ░░░██░░████████░███    ███  ░██░██ ███  ░██ ░░░░░██ ██
+░░       ░░  ░░░░░░░░ ░░░    ░░░   ░░ ░░ ░░░   ░░   █████ ░░
```
Comment (ozkatz) [inline cmd/lakefs/cmd/run.go:230]: A bit much, given our quickstart will also show this. I'd do without the big banner..

File: webui/src/lib/components/controls.jsx [hunk 2]
```diff
@@ -321,4 +321,19 @@ export const Checkbox = ({ name, onAdd, onRemove, disabled = false, defaultCheck
             }}/>
         </Form.Group>
     );
-};
\ No newline at end of file
+};
+
+export const Warning = (props) =>
+<div className="warning">
```
Comment (ozkatz) [inline webui/src/lib/components/controls.jsx:327]:
```suggestion
<Alert variant="warning">
```

File: webui/src/lib/components/navbar.jsx [hunk 3]
```diff
@@ -56,19 +55,21 @@ const TopNavLink = ({ href, children }) => {
...
-        <Navbar variant="dark" bg="dark" expand="md">
-            <Link component={Navbar.Brand} href="/">
-                <img src="/logo.png" alt="lakeFS" className="logo"/>
-            </Link>
-
-            <Nav className="mr-auto">
-                <TopNavLink href="/repositories">Repositories</TopNavLink>
-                <TopNavLink href="/auth">Administration</TopNavLink>
-            </Nav>
+	<>
```
Comment (ozkatz) [inline webui/src/lib/components/navbar.jsx:58]: why?

File: webui/src/pages/index.jsx [hunk 4]
```diff
@@ -9,21 +9,23 @@ import Setup from './setup';
...
-                </Route>
-                <Route path="/auth">
-                    <Auth/>
-                </Route>
-                <Route path="/setup">
-                    <Setup/>
-                </Route>
-            </Switch>
-        </Router>
+	<>
```
Comment (ozkatz) [inline webui/src/pages/index.jsx:12]: why?

File: webui/src/pages/setup/index.jsx [hunk 5]
```diff
@@ -108,4 +126,9 @@ server:
     );
 };
 
+const SetupPage = () => <>
+			    <SetupWarnings/>
```
Comment (ozkatz) [inline webui/src/pages/setup/index.jsx:130]: I think it looks a little out of place. At a minimum it should be on the same grid as the setup form (above it, below it, idc)

File: webui/src/styles/globals.css [hunk 6]
```diff
@@ -300,6 +300,16 @@
   font-size: 0.85rem;
 }
 
+.warning {
```
Comment (ozkatz) [inline webui/src/styles/globals.css:303]: let's use the react-bootstrap component (`<Alert variant="warning"/>`)

**Review by ozkatz — CHANGES_REQUESTED** [review]

File: webui/src/pages/setup/index.jsx (hunk 5 above)
Comment (arielshaqed) [reply webui/src/pages/setup/index.jsx]: On the "upload object" modal on the objects page:
"upload object" modal with warning about using local block adapter

**Review by arielshaqed — COMMENTED** [review]
Thanks!

PTAL (commits mostly organized and non-overlapping, you might want to read them one at a time).

**Comment by arielshaqed** [comment]
> This pull request **introduces 3 alerts** when merging [2e843ed](https://github.com/treeverse/lakeFS/commit/2e843eda4740abcb15d66bb59d72d1eedbc186b3) into [18b9afc](https://github.com/treeverse/lakeFS/commit/18b9afc2cbd9e3599fe84e9cdba66e31a44cd3f4) - [view on LGTM.com](https://lgtm.com/projects/g/treeverse/lakeFS/rev/pr-0908496908c8dd5ccd887af72f7d94074094affa)
> 
> **new alerts:**
> 
>     * 3 for Unused variable, import, function or class

These are fixed, see [new report on LGTM.com](https://lgtm.com/projects/g/treeverse/lakeFS/rev/pr-62db694679425494176ba021e2dd0c9f00a375ae).

File: api/swagger.yml [hunk 7]
```diff
@@ -681,6 +681,11 @@ components:
           type: string
         blockstore_namespace_ValidityRegex:
           type: string
+        warnings:
```
Comment (ozkatz) [inline api/swagger.yml:684]: This kind of conflicts with #2056 - I think we should fold both into a `/config/storage` endpoint and forgo the server-side warnings API (for now).

File: webui/src/styles/globals.css [hunk 8]
```diff
@@ -300,6 +300,11 @@
   font-size: 0.85rem;
 }
 
+.unmarked-list {
```
Comment (ozkatz) [inline webui/src/styles/globals.css:303]: for classes with no semantic meaning that simply want to apply a simple style, you can use Bootstrap's spacing classes (see https://getbootstrap.com/docs/4.0/utilities/spacing/#examples)

**Review by ozkatz — CHANGES_REQUESTED** [review]
I'm worried about incompatible changes with #2056

**Comment by arielshaqed** [comment]
@ozkatz , I cannot pull this because:
1. I cannot pull #2056 because that fails validation.  I did rebase on top of fix/2028-uninformative-auth-errors.
2. You did not approve this one.

Let's (finally...) close this tomorrow!

**Review by arielshaqed — COMMENTED** [review]
Thanks!

Rebased on top of `master`.  @ozkatz , PT1FL ("please take one final look")...

**Review by ozkatz — APPROVED** [review]
Looks great! thanks!

**Comment by arielshaqed** [comment]
Thanks!

**Pull request merged**